src/js_native_api_v8.cc

```diff
@@ -457,6 +457,49 @@ inline napi_status Wrap(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+// In JavaScript weak references can be created for object types (Object,
+// Function, external Object) and for local symbols that are created with
+// the `Symbol` function call. Global symbols created with `Symbol.for` method
+// cannot be weak references because they are never collected.
+// We use the `Symbol.keyFor` function call to differentiate between them. This
+// function must return `undefined` for local symbols.
+//
+// It would be more efficient to check for the internal symbol flag
+// `is_in_public_symbol_table` but it is not accessible in the public V8 API.
+// V8 has an internal function `CanBeHeldWeakly`. It could be used if it is
+// available in the public V8 API.
+inline bool CanBeHeldWeakly(napi_env env, v8::Local<v8::Value> value) {
+  if (value->IsObject()) {
+    return true;
+  }
+  if (value->IsSymbol()) {
+    v8::Isolate* isolate = env->isolate;
+    v8::Local<v8::Context> context = env->context();
+    v8::Local<v8::Function> symbol_key_for;
+    if (env->symbol_key_for_.IsEmpty()) {
+      v8::Local<v8::Object> global = context->Global();
```

Comment:
We should get `Symbol.keyFor` reference from the per-context primordial instead. See https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md for more details. The per-context primordial is available from C++ with `env->primordials()`.

Reply:
I removed this code per your recommendation.